neo4j/exceptions.py

```diff
@@ -247,15 +247,17 @@ def is_fatal_during_discovery(self) -> bool:
         return False
 
     def __str__(self):
-        return "{{code: {code}}} {{message: {message}}}".format(code=self.code, message=self.message)
+        if self.code and self.message:
+            return "{{code: {code}}} {{message: {message}}}".format(
+                code=self.code, message=self.message
+            )
+        return super().__str__()
```

Comment:
Is it possible Neo4jErro be created only with message or code?

Reply:
Shouldn't happen if the error comes from the server. In that case it goes through the `Neo4jError.hydrate` method, which makes sure both are set to some string. But this is still a valid concern for future-proofing the code.